src/private/MDMPerformerGroup.m

```diff
@@ -76,30 +78,64 @@ - (void)terminateIsActiveToken:(id<MDMIsActiveTokenable>)token
 #pragma mark - Private
 
 - (void)addPlan:(id<MDMPlan>)plan trace:(MDMTrace *)trace log:(MDMTransactionLog *)log {
-  [trace.committedPlans addObject:plan];
-  
+  // all named addPlan plans must first be removed before being added
+  if ([self isNamedTransactionLog:log] && !log.isRemoval) {
+    [self addPlan:plan trace:trace log:[[MDMTransactionLog alloc] initWithPlans:@[plan] target:log.target name:log.name removal:TRUE]];
+  }
+  // see if we can get a performer
+  id<MDMPerforming> performer = [self performerForPlan:plan trace:trace log:log];
+  if (performer != nil) {
+    if (plan != nil) {
+      [trace.committedPlans addObject:plan];
+    } else {
+      // this is the case whereby we are calling removePlan:named, but don't have a MDMPlan
```

Comment:
I think I'm being a little inconsistent here when adding/removing plans. Consider the following two cases:

**Case 1 - A very simple add named plan**
```
scheduler.addPlan(firstViewTargetAlteringPlan, named: "name_one", to: target)
// The callbacks would happen like `removePlanNamed:`, `addPlan:named`
// the trace.committedPlans would contain a removePlan, then an addPlan
```
**Case 2 - A very simple add named plan followed by a remove named plan**
```
scheduler.addPlan(firstViewTargetAlteringPlan, named: "name_one", to: target)
scheduler.removePlan(named: "name_one", from: target)
// The callbacks would happen like `removePlanNamed:`, `addPlan:named`, `removePlanNamed:`
// the trace.committedPlans would contain a removePlan, then an addPlan.
```

As `trace.committedPlans` is internal and not really accessible by the client, I'm unsure of whether this is an issue. However, the Notifications sent out might be a little confusing in this case. Basically, this unit test fails, but I'm not sure it should:
```
  func testAddAndRemoveTheSameNamedPlan() {
    let scheduler = Scheduler()
    
    var numberOfPlansCommitted = 0
    expectation(forNotification: TraceNotificationName.plansCommitted._rawValue as String, object: scheduler) { notification -> Bool in
      let event = notification.userInfo![TraceNotificationPayloadKey] as! SchedulerPlansCommittedTracePayload
      numberOfPlansCommitted = numberOfPlansCommitted + event.committedPlans.count
      // I might expect this to be true given that the callbacks executed were
      // `removePlanNamed:`, `addPlan:named`, `removePlanNamed:`
      // how could a callback get invoked without a plan being committed?
      return numberOfPlansCommitted == 3
    }
    
    scheduler.addPlan(firstViewTargetAlteringPlan, named: "name_one", to: target)
    scheduler.removePlan(named: "name_one", from: target)
    
    XCTAssertTrue(target.text! == "removePlanInvokedaddPlanInvokedremovePlanInvoked")
    
    waitForExpectations(timeout: 0.1)
  }
```
I guess we could get around this issue by storing the `name` and `target` properties on `MDMNamedPlan` and whenever we get a `removePlan:named` operation in this `else` clause, try look back through the `trace.committedPlans` for a named plan which matches the `name` and `target` that the client just provided in the `removePlanNamed:fromTarget` call in the Scheduler. Not sure if this "solution" is very elegant & unsure of whether we'd want to add these two properties to `MDMNamedPlan`

The other way around this would be to allow `MDMPerformerGroup` to keep a reference of all the committed `MDMTransactionLog` objects?